docs/api.md

```diff
@@ -0,0 +1,239 @@
+# API Documentation
+
+## Things to consider
+* An upvotes table to keep track of user upvotes
+    * Useful to prevent a single user upvoting multiple times
+* Should universities have data
+    * Flavour text ect.
+    * Some kind of reuptation or something?
+
+## Identifying Data Types
+* Users
+* Universities
+* Courses
+* Questions
+* Answers (Replies to Questions)
+* Reviews
+* Replies (Replies to Reviews)
+
+## Sample Schema's (json)
+
+### User
+```
+"user": {
+    "uid": <unique int>
+    "meta": {
+        "reputation": <int> // Merge meta/profile?
+    }
+    "username": <unqiue string>
+    "password": <string> // NOT SENT THROUGH API (db/schema)
+    "email": <unqiue string> // NOT SENT THROUGH API (db/schema)
+    "img_url" <string>
+    "profile": {
+        // Stuff for any profile data ect..
+    }
+}
+```
+
+### Universities
+```
+"university" {
+    university_id: <unqiue int>
+    "name": <string>
+    "img_url" <string>
+}
+```
+
+### Courses
+```
+"course": {
+    "course_id": <unqiue int>
+    "meta": {
+        "reputation/rating": <number>
+        "num_questions": <number>
+        "num_reviews": <number>
+    }
+    "code": <string> // e.g. COMP4920
+    "name": <string> // e.g. Ethics and Project Management
+    "univserity_id": <key> // e.g. Searching for courses by uni
+    "faculty/facult_code": e.g. "COMP" // Allows refined search
+}
+```
+
+### Questions
+```
+"question": {
+    "question_id": <unique int>
+    "meta": {
+        "user_id": <key>
+        "course_id": <key>
+        "upvotes" <number>
+        "num_answers": <number> // Can be used to determine next answer id
+    }
+    "title": <string>
+    "body": <string>
+}
+```
+
+### Answers
+```
+"answer": {
+    "answer_id": <unique int>
+    "meta": {
+        "user_id": <key>
+        "question_id": <key>
+        "upvotes" <number>
+    }
+    "title": <string>
+    "body": <string>
+}
+```
+
+### Reviews
+```
+"review": {
+    "review_id": <unique int>
+    "meta": {
+        "user_id": <key>
+        "course_id": <key>
+        "upvotes" <number>
+        "num_replies": <number> // Can be used to determine next reply id
+    }
+    "title": <string>
+    "body": <string>
+}
+```
+
+### Replies
```

Comment:
Wonder whether having replies as their own schema is useful. Might just add complexity. Is there anything any way a reply differs from an answer?